src/workerd/api/crypto/crypto.c++

```diff
@@ -676,13 +689,61 @@ kj::String Crypto::randomUUID() {
 // =======================================================================================
 // Crypto Streams implementation
 
+class CRC32DigestContext: public DigestContext {
+ public:
+  CRC32DigestContext(): value(crc32(0, Z_NULL, 0)) {}
+  virtual ~CRC32DigestContext() = default;
+
+  void write(kj::ArrayPtr<kj::byte> buffer) {
+    value = crc32(value, buffer.begin(), buffer.size());
+  }
+
+  kj::Array<kj::byte> close() {
+    auto beValue = htonl(value);
+    return kj::heapArray<kj::byte>((kj::byte*)&beValue, sizeof(value));
```

Comment:
Should be sizeof(beValue) though I could also statically assert these sizes are equal